test/integration/TokenAirdropIntegrationTest.js

```diff
@@ -0,0 +1,538 @@
+
+        const { serials } = await mintResponse.getReceipt(env.client);
+
+        const receiverPrivateKey = PrivateKey.generateED25519();
+        const receiverPublicKey = receiverPrivateKey.publicKey;
+        const accountCreateResponse = await (
+            await new AccountCreateTransaction()
+                .setKey(receiverPublicKey)
+                .setReceiverSignatureRequired(true)
+                .freezeWith(env.client)
+                .sign(receiverPrivateKey)
+        ).execute(env.client);
+
+        const { accountId: receiverId } =
+            await accountCreateResponse.getReceipt(env.client);
+
+        let err = false;
+        try {
+            const airdropTokenResponse = await new TokenAirdropTransaction()
+                .addTokenTransfer(tokenId, receiverId, INITIAL_SUPPLY)
+                .addTokenTransfer(tokenId, env.operatorId, -INITIAL_SUPPLY)
+                .addNftTransfer(
+                    nftTokenId,
+                    serials[0],
+                    env.operatorId,
+                    receiverId,
+                )
+                .execute(env.client);
+
+            await airdropTokenResponse.getReceipt(env.client);
+        } catch (error) {
+            if (error.message.includes("INVALID_SIGNATURE")) {
+                err = true;
+            }
+        }
+
+        expect(err).to.be.eq(true);
+    });
+
+    it("should not airdrop with invalid tx body", async function () {
+        let err = false;
+        const tokenId = new TokenId(1);
+        const accountId = new AccountId(1);
+
+        try {
+            await (
+                await new TokenAirdropTransaction().execute(env.client)
+            ).getReceipt(env.client);
+        } catch (error) {
+            // SHOULD IT FAIL WITH INVALID TX BODY
+            if (error.message.includes("FAIL_INVALID")) {
+                err = true;
+            }
+        }
+        expect(err).to.be.eq(true);
+
+        err = false;
+        try {
+            await (
+                await new TokenAirdropTransaction()
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addTokenTransfer(tokenId, accountId, 1)
+                    .addNftTransfer(new NftId(tokenId, 1), accountId, accountId)
+                    .execute(env.client)
+            ).getReceipt(env.client);
+        } catch (error) {
+            if (error.message.includes("INVALID_TRANSACTION_BODY")) {
+                err = true;
```

Comment:
For this scenario we would need 10+ tokens, not 10+ transfers, so i would suggest to remove this since it would be quite slow and chunky.

Reply:
Creating 10 different tokens is going to be slow too. This basically accomplishes the error without having to create 10 different tokens.

Reply:
Oh, so this test is passing?

Reply:
yep